### PR TITLE
Update reboot exception documentation and add GPG

### PR DESCRIPTION
**Pull request by mkenigs**
GPG keys were added to the list of reboot exceptions, and inaccurate
documentation was updated. Selected registries.conf changes were
previously listed under the "None" action, but registries.conf changes
always trigger the "Reload Crio" action. Some organizational and wording
changes were made to make this more clear

Depends https://github.com/openshift/machine-config-operator/pull/2851

### CODE REVIEW

**Comment by kikisdeliveryservice** [comment]
just a hold until #2855 2851 merges

File: docs/MachineConfigDaemon.md [hunk 1]
```diff
@@ -146,27 +146,38 @@ Etcd is co-located on master nodes as static pods. The draining behavior defined
...
-   - appending items in unqualified-search-registries list
+1. [SSH Keys](./Update-SSHKeys.md): updated by changing `ignition.passwd.users.sshAuthorizedKeys` in a MachineConfig
+2. kube-apiserver-to-kubelet-signer CA cert: located at `/etc/kubernetes/kubelet-ca.crt` and autorotated by the openshift-kubeapiserver operator after a 1 year expiry
+3. [Pull Secret](./PullSecret.md): cluster-wide, located at `/var/lib/kubelet/config.json`
 
-### Rebootless Updates
+### "Reload Crio" Action
 
-"Crio Reload" action: performs the file write, and runs a `systemctl reload crio`. The following changes will trigger a drain, but not a reboot:
+The "Reload Crio" action performs the file write and runs a `systemctl reload crio`. It does not trigger a reboot, but it may trigger a drain.
```
Comment (kikisdeliveryservice) [inline docs/MachineConfigDaemon.md:163]:
```suggestion
The "Reload Crio" action performs the file write and runs a `systemctl reload crio`. It does not trigger a reboot, but may trigger a drain:
```

File: docs/MachineConfigDaemon.md [hunk 2]
```diff
@@ -146,27 +146,38 @@ Etcd is co-located on master nodes as static pods. The draining behavior defined
 
 ## Optimized Updates
 
-As of Openshift 4.7, the MCD gained the functionality to apply select MachineConfig updates without a full reboot flow (drain -> update -> reboot). The action is calculated as a diff between current and desired configurations. For any MachineConfig change not listed below, or if a forcefile was set, the MCD will trigger the full reboot flow.
+As of Openshift 4.7, the MCD gained the functionality to apply select MachineConfig updates without a full reboot flow (drain -> update -> reboot). The MCD now calculates a diff between the current and desired configurations, and it uses any changes to select one of the actions listed below. For any change not listed below, or if a forcefile was set, the MCD will trigger the full reboot flow.
```
Comment (kikisdeliveryservice) [inline docs/MachineConfigDaemon.md:149]:
```suggestion
As of Openshift 4.7, the MCD gained the functionality to apply select MachineConfig updates without a full reboot flow (drain -> update -> reboot). The MCD now calculates a diff between the current and desired configurations, and it uses any changes to select one of the options listed below. For any change not listed below, or if a forcefile was set, the MCD will trigger the full reboot flow.
```

**Review by kikisdeliveryservice — COMMENTED** [review]
A few comments, thanks for making a companion doc update pr!

File: docs/MachineConfigDaemon.md [hunk 3]
```diff
@@ -146,27 +146,38 @@ Etcd is co-located on master nodes as static pods. The draining behavior defined
...
 
-The updated list of optimized updates and behaviour (as of Openshift 4.8) is as follows:
+The updated list of optimized updates and behaviour (as of Openshift 4.10) is as follows:
 
-### Drainless and Rebootless Updates
+### "None" Action
```
[the remark this answers is not in the document]
Comment (mkenigs) [reply docs/MachineConfigDaemon.md]: I reorganized it as
Rebootless
Without Drain or With Drain
None or Reload Crio

Is that good? I can leave the headers exactly as they were, but then reload crio and none will have to be lumped together

[the remark this answers is not in the document]
Comment (kikisdeliveryservice) [reply docs/MachineConfigDaemon.md]: I think this works now :)

**Comment by kikisdeliveryservice** [comment]
> just a hold until #2851 merges

apparently i forgot to add the actual hold :laughing: 

/hold

**Comment by yuqi-zhang** [comment]
/hold cancel

I think this looks good and the other PR has merged. Do you have any updates @mkenigs ?

**Comment by mkenigs** [comment]
Yep should be good to go now that the other one merged!

**Review by yuqi-zhang — APPROVED** [review]
/lgtm

**Comment by sinnykumari** [comment]
Adding valid bz as this a doc update, so doesn't need a real bug created

**Pull request merged**